specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-08-02-preview/managedClusters.json

```diff
@@ -3556,9 +3556,87 @@
         "clientId"
       ]
     },
+    "FlexProfile": {
+      "type": "object",
+      "properties": {
+        "scaleProfile": {
+          "$ref": "#/definitions/ScaleProfile",
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      },
+      "description": "Specifications on flex Agentpool."
+    },
+    "ScaleProfile": {
+      "type": "object",
+      "properties": {
+        "manual": {
+          "$ref": "#/definitions/ManualScaleProfile",
+          "description": "Specifications on how to scale the flex agent pool to a fixed size."
+        }
+      },
+      "description": "Specifications on how to scale the AgentPool."
+    },
+    "ManualScaleProfile": {
+      "type": "object",
+      "properties": {
+        "machines": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManualScaleMachineSpec"
+          },
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      }
+    },
+    "ManualScaleMachineSpec": {
+      "type": "object",
+      "properties": {
+        "sizes": {
+          "items": {
+            "type": "string"
+          },
+          "description": "The comma-separated list allowed vm sizes. AKS will use the first available one upon scale."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of nodes (VMs). Allowed values must be in the range of 0 to 1000 (inclusive)."
+        }
+      },
+      "description": "Specifications on how to scale the AgentPool."
+    },
+    "CurrentFlexNodes": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/flexNodes"
+      },
+      "x-ms-identifiers": [],
+      "description": "The manual scale specifications of flexVirtualMachines AgentPool."
+    },
+    "FlexNodes": {
+      "type": "object",
+      "properties": {
+        "sizes": {
+          "type": "string",
+          "description": "The VM size of the agents used to host this group of nodes."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of nodes."
+        }
+      },
+      "description": "Current status on a group of nodes of the same vm size."
+    },
     "ManagedClusterAgentPoolProfileProperties": {
       "type": "object",
       "properties": {
+        "flexProfile": {
+          "$ref": "#/definitions/FlexProfile"
+        },
+        "currentFlexNodes": {
+          "$ref": "#/definitions/CurrentFlexNodes"
```

Comment:
minor, naming: This isn't really _nodes_ it's a nodes summary (it just contains the count).

I don't know if there's a better name than what you have for it, but in some ways saying `nodes` here makes you think "oh this will be a collection with 1 item per node" when it's not.